### PR TITLE
Add an option to add Modpacks from the Game menu

### PR DESCRIPTION
**Pull request by jwrober**
Also fix two issues where the client would not refresh when content on disk changed after the client was opened. This allows the client to see changes after content is downloaded with the modpack installer. See #2158

### CODE REVIEW

File: client/audio/audio.cpp [hunk 1]
```diff
@@ -78,40 +78,26 @@ const QVector<QString> *get_soundplugin_list(const struct option *poption)
...
    Returns a static string vector of soundsets available on the system.
  */
 const QVector<QString> *get_soundset_list(const struct option *poption)
 {
-  static QVector<QString> *sound_list = nullptr;
-
-  return get_audio_speclist(SNDSPEC_SUFFIX, &sound_list);
+  static QVector<QString> *sound_list = new QVector<QString>;
+  sound_list->clear();
+  *sound_list = *fileinfolist(get_data_dirs(), SNDSPEC_SUFFIX);
```
Comment (lmoureaux) [inline client/audio/audio.cpp:88]: We need to `delete` the pointer returned by `fileinfolist` or we'll have a memory leak. In addition, clearing isn't needed before an assignment.
```suggestion
  auto list = fileinfolist(get_data_dirs(), SNDSPEC_SUFFIX);
  *sound_list = std::move(*list);
  delete list;
```

File: client/audio/audio.cpp [hunk 2]
```diff
@@ -78,40 +78,26 @@ const QVector<QString> *get_soundplugin_list(const struct option *poption)
...
    Returns a static string vector of musicsets available on the system.
  */
 const QVector<QString> *get_musicset_list(const struct option *poption)
 {
-  static QVector<QString> *music_list = nullptr;
-
-  return get_audio_speclist(MUSICSPEC_SUFFIX, &music_list);
+  static QVector<QString> *music_list = new QVector<QString>;
+  music_list->clear();
+  *music_list = *fileinfolist(get_data_dirs(), MUSICSPEC_SUFFIX);
```
Comment (lmoureaux) [inline client/audio/audio.cpp:99]: We need to `delete` the pointer returned by `fileinfolist` or we'll have a memory leak. In addition, clearing isn't needed before an assignment.
```suggestion
  auto list = fileinfolist(get_data_dirs(), MUSICSPEC_SUFFIX);
  *music_list = std::move(*list);
  delete list;
```

File: client/tileset/tilespec.cpp [hunk 3]
```diff
@@ -774,31 +774,29 @@ QEvent::Type TilesetChanged;
...
-    for (const auto &file : qAsConst(*list)) {
-      struct tileset *t =
-          tileset_read_toplevel(qUtf8Printable(file), false, topo);
-
-      if (t) {
-        tilesets[idx]->append(file);
-        tileset_free(t);
-      }
+  /* Note: this means you must restart the client after installing a new
+      tileset. */
```
Comment (lmoureaux) [inline client/tileset/tilespec.cpp:787]: This comment is hopefully no longer valid :sweat_smile: 
```suggestion
```

File: client/tileset/tilespec.cpp [hunk 4]
```diff
@@ -774,31 +774,29 @@ QEvent::Type TilesetChanged;
...
+  QVector<QString> *list = fileinfolist(get_data_dirs(), TILESPEC_SUFFIX);
+  tilesets[idx]->clear();
+  tilesets[idx] = new QVector<QString>;
```
Comment (lmoureaux) [inline client/tileset/tilespec.cpp:790]: We don't need to allocate a new vector if we `clear`ed it before.
```suggestion
```

**Review by lmoureaux — CHANGES_REQUESTED** [review]
With this change, I think it would make sense to forbid building the client without the mod installer. We should probably check our build options in CMake.

**Comment by lmoureaux** [comment]
(We can do the CMake changes in another PR, but let's not introduce memory leaks)

File: client/audio/audio.cpp (hunk 1 above)
Comment (jwrober) [reply client/audio/audio.cpp:88]: I figured a `clear()` before using it did the same. I'll fix.

**Review by lmoureaux — APPROVED** [review]
Nice work!

**Pull request merged**